### ISSUE TITLE
Package manager is not available

### ISSUE THREAD

**Issue by fiLLLip** [meanmail-dev/requirements#47]
**Describe the bug**
Plugin says package manager is not available.

**To Reproduce**
1. Have a workspace with multiple projects, both Python an non-python
2. Set up Python virtual env for each python project
3. See requirements.txt file in project
4. Says package not installed
5. Click install
6. Error with "Package manage is not available"

`11:23 AM	Installing 'black~=20.8b1': Package manager is not available`

**Expected behavior**
Should show packages as installed and 

**IDE (please complete the following information. See Main Menu -> Help -> About):**
 - OS: [PopOS 20.04 LTS]
 - Product: [IntelliJ IDEA 2020.2.2 (Ultimate Edition)]
 - Version [IU-202.7319.50]
 - JRE Version: [OpenJDK 64-Bit Server VM by JetBrains s.r.o.]
 - Project SDK: [Python 3.7.9]

**Plugin (please complete the following information. See Main Menu -> File -> Settings -> Plugins):**
 - Version [2020.2-202]

**Additional context**
```
java.lang.Throwable: ~=20.8b1
 parent=~=20.8b1
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:143)
	at com.intellij.psi.impl.PsiElementBase.notNullChild(PsiElementBase.java:284)
	at com.intellij.psi.impl.PsiElementBase.findNotNullChildByClass(PsiElementBase.java:307)
	at ru.meanmail.psi.impl.VersionOneImpl.getVersion(VersionOneImpl.kt:18)
	at ru.meanmail.psi.impl.VersionOneImpl.getReference(VersionOneImpl.kt:43)
	at com.intellij.psi.impl.SharedPsiElementImplUtil.getReferences(SharedPsiElementImplUtil.java:76)
	at com.intellij.psi.impl.PsiElementBase.getReferences(PsiElementBase.java:81)
	at com.intellij.psi.impl.SharedPsiElementImplUtil.addReferences(SharedPsiElementImplUtil.java:59)
	at com.intellij.psi.impl.SharedPsiElementImplUtil.findReferenceAt(SharedPsiElementImplUtil.java:33)
	at com.intellij.psi.impl.SharedPsiElementImplUtil.findReferenceAt(SharedPsiElementImplUtil.java:51)
	at com.intellij.psi.impl.PsiElementBase.findReferenceAt(PsiElementBase.java:86)
	at com.intellij.psi.AbstractFileViewProvider.findReferenceAt(AbstractFileViewProvider.java:240)
	at com.intellij.psi.SingleRootFileViewProvider.findReferenceAt(SingleRootFileViewProvider.java:224)
	at com.intellij.psi.impl.source.PsiFileImpl.findReferenceAt(PsiFileImpl.java:529)
	at com.intellij.codeInsight.TargetElementUtilBase.findReference(TargetElementUtilBase.java:189)
	at com.intellij.codeInsight.TargetElementUtil.findReference(TargetElementUtil.java:93)
	at com.intellij.model.psi.impl.TargetsKt.fromTargetEvaluator(targets.kt:158)
	at com.intellij.model.psi.impl.TargetsKt.declarationsOrReferences(targets.kt:127)
	at com.intellij.model.psi.impl.TargetsKt.declaredReferencedData(targets.kt:34)
	at com.intellij.model.psi.impl.TargetsKt.targetSymbols(targets.kt:24)
	at com.intellij.codeInsight.daemon.impl.IdentifierHighlighterPass.getTargetSymbols(IdentifierHighlighterPass.java:233)
	at com.intellij.codeInsight.daemon.impl.IdentifierHighlighterPass.highlightReferencesAndDeclarations(IdentifierHighlighterPass.java:225)
	at com.intellij.codeInsight.daemon.impl.IdentifierHighlighterPass.doCollectInformation(IdentifierHighlighterPass.java:90)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:54)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:399)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1110)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:392)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:629)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:581)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:391)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:367)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:170)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:182)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:365)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:181)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by arnaldop** [comment]
I am experiencing this problem also, although I have not seen the exception shown above.

The issue is that [ru.meanmail.psi.RequirementsPsiImplUtil.kt](https://github.com/meanmail-dev/requirements/blob/master/src/main/kotlin/ru/meanmail/psi/RequirementsPsiImplUtil.kt) getSdk() checks if the root project's primary SDK is a Python SDK. If the project's primary SDK is not Python (in my case, it's a Java project, with a sub-project in Python), then this test will fail.

Alternatives would be:
1. To cycle through the available SDKs, and if there is only a single Python SDK, choose it.
1. To attempt to identify the IntelliJ Project Sub-Module that contains the given `requirements.txt` (based on the paths, perhaps), and then check if that sub-module's SDK is a Python SDK.
1. In my `.idea/<main-project>.iml` I have a Python facet, which includes the Python SDK name. Perhaps a reference to the SDK can be inferred or derived from this.

**Comment by balpeaux** [comment]
Hello,

I have the same problem, however, I'm not mixing Java and Python modules.
My project is **exclusively python 3.8**.

Hope it gets fixed soon, cause it's a great plugin.
Thanks by the way!

**Comment by jmesterh** [comment]
Also seeing this bug, with a Java maven project. It has a Python module, which is using a virtualenv as the Python SDK.

**Comment by meanmail** [comment]
Fixed in 2020.5. New version available for download

**Comment by dougbergh** [comment]
I am having this problem, running version 2021.4.1-202
